src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTaskTest.cpp

```diff
@@ -0,0 +1,126 @@
+
+#include <QFileInfo>
+
+#include <U2Core/DNASequenceObject.h>
+#include <U2Core/DocumentModel.h>
+#include <U2Core/U2SafePoints.h>
+
+namespace U2 {
+
+#define DOC_NAME_ATTR "docName"
+#define SEQ_NAME_ATTR "seqName"
+#define ENERGY_ATTR "gibbsFreeEnergy"
+#define MELTING_POINT_ATTR "meltingPoint"
+#define OVERLAP_ATTR "overlapLength"
+#define ENERGY_EXCLUDE_ATTR "gibbsFreeEnergyExclude"
+#define MELTING_POINT_EXCLUDE_ATTR "meltingPointExclude"
+#define COMPL_LENGTH_EXCLUDE_ATTR "complementLengthExclude"
+#define INSERT_TO_ATTR "insertTo"
+#define LEFT_AREA_ATTR "leftArea"
+#define RIGHT_AREA_ATTR "rightArea"
+#define RESULT_ATTR "expectedResultsList"
+#define BACKBONE_URL_ATTR "backboneSequenceUrl"
+#define OTHER_SEQ_PCR_URL_ATTR "otherSequencesInPcrUrl"
+
+void GTest_PCRPrimerDesignForDNAAssemblyTaskTest::init(XMLTestFormat *, const QDomElement &el) {
+    docName = el.attribute(DOC_NAME_ATTR);
+    seqName = el.attribute(SEQ_NAME_ATTR);
+    settings.gibbsFreeEnergy = getU2RangeInt(el, ENERGY_ATTR, ";");
+    settings.meltingPoint = getU2RangeInt(el, MELTING_POINT_ATTR, ";");
+    settings.overlapLength = getU2RangeInt(el, OVERLAP_ATTR, ";");
+    settings.gibbsFreeEnergyExclude = getInt(el, ENERGY_EXCLUDE_ATTR);
+    settings.meltingPointExclude = getInt(el, MELTING_POINT_EXCLUDE_ATTR);
+    settings.complementLengthExclude = getInt(el, COMPL_LENGTH_EXCLUDE_ATTR);
+    QString buf = el.attribute(INSERT_TO_ATTR);
+    if (buf == "5") {
+        settings.insertTo = PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone5;
+    } else if (buf == "3") {
+        settings.insertTo = PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone3;
+    } else {
+        wrongValue(INSERT_TO_ATTR);
+        return;
+    }
+    settings.leftArea = getU2Region(el, LEFT_AREA_ATTR);
+    settings.rightArea = getU2Region(el, RIGHT_AREA_ATTR);
+    result = getU2RegionList(el, RESULT_ATTR, ";", 8);
+    buf = el.attribute(BACKBONE_URL_ATTR);
+    if (!buf.isEmpty()) {
+        QFileInfo refFile(env->getVar("COMMON_DATA_DIR") + "/" + buf);
+        if (!refFile.exists()) {
+            stateInfo.setError(QString("file not exist %1").arg(refFile.absoluteFilePath()));
+            return;
+        }
+        settings.backboneSequenceUrl = env->getVar("COMMON_DATA_DIR") + "/" + buf;
+    }
+    buf = el.attribute(OTHER_SEQ_PCR_URL_ATTR);
+    if (!buf.isEmpty()) {
+        QFileInfo refFile(env->getVar("COMMON_DATA_DIR") + "/" + buf);
+        if (!refFile.exists()) {
+            stateInfo.setError(QString("file not exist %1").arg(refFile.absoluteFilePath()));
+            return;
+        }
+        settings.otherSequencesInPcrUrl = env->getVar("COMMON_DATA_DIR") + "/" + buf;
+    }
+}
+
+void GTest_PCRPrimerDesignForDNAAssemblyTaskTest::prepare() {
+    Document *loadedDocument = getContext<Document>(this, docName);
+    CHECK_EXT(loadedDocument != nullptr, stateInfo.setError(GTest::tr("context not found %1").arg(docName)), );
```

Comment:
Is translation required here? The same 3 lines below.

Reply:
No, there is not need in translation of tests code.